src/storage/backend/redis/Redis.cc

```diff
@@ -388,37 +388,40 @@ void Redis::DoExpire(double current_network_time) {
         return;
     }
 
-    std::vector<std::string> elements;
-    for ( size_t i = 0; i < reply->elements; i++ )
-        elements.emplace_back(reply->element[i]->str);
+    std::vector<std::string_view> elements;
+    elements.reserve(reply->elements);
 
-    freeReplyObject(reply);
+    for ( size_t i = 0; i < reply->elements; i++ )
+        elements.emplace_back(reply->element[i]->str, reply->element[i]->len);
 
     // TODO: it's possible to pass multiple keys to a DEL operation but it requires
     // building an array of the strings, building up the DEL command with entries,
     // and passing the array as a block somehow. There's no guarantee it'd be faster
     // anyways.
     for ( const auto& e : elements ) {
-        status = redisAsyncCommand(async_ctx, redisGeneric, NULL, "DEL %s:%s", key_prefix.data(), e.c_str());
+        status = redisAsyncCommand(async_ctx, redisGeneric, NULL,
+                                   util::fmt("DEL %s:%.*s", key_prefix.data(), static_cast<int>(e.size()), e.data()));
```

Comment:
would it perhaps make sense to put a comment here why util::fmt is being used? Similar to the commit message. Otherwise this might be confusing to potential future reviewers.